### PR TITLE
Use project path instead of working directory to resolve launchSettings.json

### LINKED ISSUE #35393: dotnet-watch: Blazor Hosted project specifying Server path via --project does not work
Repro steps:

```
dotnet new blazorwasm --hosted -o blazorhosted
code .
dotnet watch  -- project .\Server\blazorhosted.Server.csproj
```

### PR DESCRIPTION
**Pull request by pranavkm**
Fixes https://github.com/dotnet/aspnetcore/issues/35393

dotnet-watch attempts to read the app's launchSettings.json to read things like the launch URL. Currently the URL is read relative to the working directory (it looks for `$WorkDir/Properties/launchSettings.json`). However, this is incorrect if the `--project` option is used to specify a project path / directory. This change updates the code to use the project directory and adds a couple of tests.

### CODE REVIEW

File: src/Tests/dotnet-watch.Tests/Utilities/WatchableApp.cs [hunk 1]
```diff
@@ -76,7 +78,7 @@ public void Start(IEnumerable<string> arguments, [CallerMemberName] string name
 
             var commandSpec = new DotnetCommand(_logger, args.ToArray())
             {
-                WorkingDirectory = SourceDirectory,
+                WorkingDirectory = WorkingDirectory ?? SourceDirectory,
```
Comment (TanayParikh) [inline src/Tests/dotnet-watch.Tests/Utilities/WatchableApp.cs:81]:
```suggestion
                WorkingDirectory ??= SourceDirectory,
```

**Review by TanayParikh — APPROVED** [review]

**Comment by TanayParikh** [comment]
Looks like the CI isn't happy yet though 👀

File: src/Tests/dotnet-watch.Tests/Utilities/WatchableApp.cs (hunk 1 above)
Comment (pranavkm) [reply src/Tests/dotnet-watch.Tests/Utilities/WatchableApp.cs:81]: That wouldn't work - we're assigning the `WorkingDirectory` property from the current instance to the `DotNetCommand` instance.

**Review by TanayParikh — APPROVED** [review]
LGTM, CI is green

**Pull request merged**